### PR TITLE
MNT: Run PTH flake test in prep for supporting pathlib (modeling)

### PR DESCRIPTION
**Pull request by neutrinoceros**
### Description
Ref #16924
This is in the continuation of #16060

- [ ] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

**Comment by neutrinoceros** [comment]
Oh, good catch, I missed that these were targeting the same sub-package !
The answer is (mostly) no: the present PR is minimalist but #16927 contains other valuable work from @MridulS for the test sub directory (`test_*` files have their own set of ignored rules in `.ruff.toml`, so it can be done separately).
I would recommend merging this one first, as I need to come back to #16927 later anyway.

File: astropy/modeling/fitting_parallel.py
Comment (pllim) [inline astropy/modeling/fitting_parallel.py:1]: Can `f.write` be replaced with `write_text`?

Comment (neutrinoceros) [reply astropy/modeling/fitting_parallel.py:1]: you got it !

**Review by pllim — APPROVED** [review]
Diff looks fine. Since this is generic path handling not related to actual fitting algorithm, I think I am comfortable enough to approve myself. Let's wait for the CI though.

Thanks!

**Comment by pllim** [comment]
Hmm. I see this error

```
      File "./astropy/table/setup_package.py", line 17, in get_extensions
        Extension(
      File "/usr/lib/python3/dist-packages/setuptools/extension.py", line 130, in __init__
        super().__init__(name, sources, *args, **kw)
      File "/usr/lib/python3/dist-packages/setuptools/_distutils/extension.py", line 110, in __init__
        raise AssertionError("'sources' must be a list of strings")
    AssertionError: 'sources' must be a list of strings
```

but it is from https://github.com/astropy/astropy/pull/16939 ?

**Comment by astrofrog** [comment]
@neutrinoceros - could you rebase?

**Comment by neutrinoceros** [comment]
@astrofrog done !

**Comment by pllim** [comment]
Thanks!

**Pull request merged**